### PR TITLE
LPS-191800 | Test Automation Creation | Master

### PR DESCRIPTION
**Pull request by rafaellajordao**
# LPS-191800 | Test Automation Creation

Hi! :)
This pr is corresponding to this ticket: [LPS-191007](https://liferay.atlassian.net/browse/LPS-191007) → [LPS-191800](https://liferay.atlassian.net/browse/LPS-191800)

### CODE REVIEW

**Comment by rafaellajordao** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase [hunk 1]
```diff
@@ -190,6 +190,121 @@ element.textContent = 'Fragment Example';''';
...
+				JSONBlog.addEntry(
+					entryContent = "Blog Test Content ${i}",
+					entryTitle = "Blog Test Title ${i}",
+					groupName = "Test Site Name");
+			}
+
+			JSONAssetlist.addDynamicAssetListEntryWithItemType(
+				groupName = "Test Site Name",
+				itemType = "All Types",
+				title = "Dynamic Collection");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase:261]: Use UI steps that is faster than this JSON macro

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase [hunk 2]
```diff
@@ -190,6 +190,121 @@ element.textContent = 'Fragment Example';''';
...
+
+			LexiconCard.clickCard(card = "Dynamic Collection");
+		}
+
+		task ("Then the fragment developer should be able to see a limited number of asset") {
+			AssertVisible(
+				key_entryContent = "[normalize-space(text())='Blog Test Title 3']",
+				key_fragmentName = "Fragment Name",
+				key_position = 1,
+				locator1 = "PageEditor#COLLECTION_ITEM_CONTENT_IN_BULLETED_LIST_AT_VIEW_MODE");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase:292]: COLLECTION_ITEM_CONTENT

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase [hunk 3]
```diff
@@ -190,6 +190,121 @@ element.textContent = 'Fragment Example';''';
...
+
+			AssertVisible(
+				key_entryContent = "[normalize-space(text())='Blog Test Title 2']",
+				key_fragmentName = "Fragment Name",
+				key_position = 2,
+				locator1 = "PageEditor#COLLECTION_ITEM_CONTENT_IN_BULLETED_LIST_AT_VIEW_MODE");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase:298]: COLLECTION_ITEM_CONTENT

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase [hunk 4]
```diff
@@ -190,6 +190,121 @@ element.textContent = 'Fragment Example';''';
...
+
+			AssertElementNotPresent(
+				key_entryContent = "[normalize-space(text())='Blog Test Title 1']",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase:301]: ""

**Review by Tim-Cao — CHANGES_REQUESTED** [review]
Hi @rafaellajordao, could you modify according to comments? Thanks

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase [hunk 5]
```diff
@@ -190,6 +190,121 @@ element.textContent = 'Fragment Example';''';
...
+
+			PageEditor.gotoConfiguration(fragmentName = "Fragment Name");
+
+			Click(locator1 = "Button#PLUS");
+
+			SelectFrame(locator1 = "IFrame#MODAL_BODY");
+
+			NavItem.click(navItem = "Collections");
+
+			LexiconCard.clickCard(card = "Dynamic Collection");
```
[the remark this answers is not in the document]
Comment (rafaellajordao) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase:284]: didn't work

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase (hunk 3 above)
Comment (rafaellajordao) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase:298]: didn't work

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase (hunk 2 above)
Comment (rafaellajordao) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase:292]: didn't work

**Comment by rafaellajordao** [comment]
ci:test:sf

**Comment by Tim-Cao** [comment]
See changes @rafaellajordao https://github.com/liferay-echo/liferay-portal/pull/13202

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**